db/Pos/DB/Class.hs

```diff
@@ -104,13 +113,29 @@ class (HasConfiguration, MonadBaseControl IO m, MonadThrow m) => MonadDBRead m w
         , Bi (IterValue i)
         ) => DBTag -> Proxy i -> Source (ResourceT m) (IterType i)
 
+    -- | Get block by header hash
+    dbGetRawBlock :: HeaderHash -> m (Maybe RawBlock)
+
+    -- | Get undo by header hash
+    dbGetRawUndo :: HeaderHash -> m (Maybe RawUndo)
+
 instance {-# OVERLAPPABLE #-}
     (MonadDBRead m, MonadTrans t, MonadThrow (t m), MonadBaseControl IO (t m)) =>
         MonadDBRead (t m)
   where
     dbGet tag = lift . dbGet tag
     dbIterSource tag (p :: Proxy i) =
         hoist (hoist lift) (dbIterSource tag p)
+    dbGetRawBlock = lift . dbGetRawBlock
+    dbGetRawUndo = lift . dbGetRawUndo
+
+
+type MonadBlockDBRead m = (MonadDBRead m, BlockchainHelpers MainBlockchain)
+
+getBlock :: MonadBlockDBRead m => HeaderHash -> m (Maybe Block)
+getBlock x = do
+    mBS <- fmap unRawBlock <$> dbGetRawBlock x
+    pure $ rightToMaybe . decodeFull =<< mBS
```

Comment:
Why do you discard decode error? `Nothing` means that there is no block. Decode error should trigger `DBMalfored` exception.